rmm/monitor/src/host/mod.rs

```diff
@@ -0,0 +1,19 @@
+pub mod pointer;
+
+use crate::rmm::PageMap;
+
+/// This trait is used to enforce security checks for physical region allocated by the host.
+pub trait Accessor {
+    /// Try to do page-relevant stuff (e.g., RTT map).
+    /// returns true only if everything goes well.
+    fn acquire(ptr: usize, page_map: PageMap) -> bool;
+
+    /// Try to clean up page-relevant stuff done by `acquire`.
+    /// Structs that implement this trait must synchronize this function with `acquire`.
+    /// returns true only if everything goes well.
+    fn release(ptr: usize, page_map: PageMap) -> bool;
```

Comment:
How about add default implementations? I think most structs are similar.

```suggestion
    fn acquire(ptr: usize, page_map: PageMap) -> bool {
            page_map.map(ptr, false);
    }

    /// Try to clean up page-relevant stuff done by `acquire`.
    /// Structs that implement this trait must synchronize this function with `acquire`.
    /// returns true only if everything goes well.
    fn release(ptr: usize, page_map: PageMap) -> bool {
        page_map.unmap(ptr)
    }
```

Reply:
Nice suggestion, I'll reflect that right away

Reply:
Reflected: